src/test/java/com/redhat/foreman/ForemanSharedNodeCloudTest.java

```diff
@@ -82,112 +104,42 @@
      * Rule for wiremock.
      */
     @Rule
-    public final WireMockRule wireMockRule = new WireMockRule(32789);
+    public final WireMockRule wireMockRule = new WireMockRule(SERVICE_PORT);
+
+    // Use per-test instance rather than the static helper methods as they are talking to thread local instance so we
+    // can not configure stubs from different threads.
+    private WireMock wireMock;
+    private StubMapping stubFor(MappingBuilder mapping) {
+        if (wireMock == null) {
+            wireMock = new WireMock(SERVICE_PORT);
+        }
+        StubMapping build = mapping.build();
+        wireMock.register(build);
+        return build;
+    }
 
     /**
      * Prepare wiremocks.
-     *
-     * @throws IOException if occurs.
-     * @throws URISyntaxException if occurs.
      */
+    // TODO: Avoid shared fixture
     private void setupWireMock() throws IOException, URISyntaxException {
-        String body1  = TestUtils.readFile("body1.txt", Charset.forName("UTF-8"));
-        String body2  = TestUtils.readFile("body2.txt", Charset.forName("UTF-8"));
-        String body3  = TestUtils.readFile("body3.txt", Charset.forName("UTF-8"));
-        String body4  = TestUtils.readFile("body4.txt", Charset.forName("UTF-8"));
-        String body5  = TestUtils.readFile("body5.txt", Charset.forName("UTF-8"));
-        String body6  = TestUtils.readFile("body6.txt", Charset.forName("UTF-8"));
-        String body7  = TestUtils.readFile("body7.txt", Charset.forName("UTF-8"));
-        String body8  = TestUtils.readFile("body8.txt", Charset.forName("UTF-8"));
-        String body9  = TestUtils.readFile("body9.txt", Charset.forName("UTF-8"));
-        String body10 = TestUtils.readFile("body10.txt", Charset.forName("UTF-8"));
-        String body11 = TestUtils.readFile("body11.txt", Charset.forName("UTF-8"));
-
         stubFor(get(urlEqualTo("/api/v2/hosts?search=params.JENKINS_LABEL%3Dlabel1"))
-                .willReturn(aResponse()
-                        .withStatus(HTTPOK)
-                        .withHeader("Content-Type", "text/json")
-                        .withBody(body1)));
-
-        stubFor(get(urlEqualTo("/api/v2/hosts?search=params.JENKINS_LABEL%3Dlabel1+and+params.RESERVED%3D"
-                + "false+and+has+params.JENKINS_SLAVE_REMOTEFS_ROOT"))
-                .willReturn(aResponse()
-                        .withStatus(HTTPOK)
-                        .withHeader("Content-Type", "text/json")
-                        .withBody(body2)));
+                .willReturn(ok("body1.txt")));
```

Comment:
I have only named those I needed. The method and unrenamed resources are hereby unused and kept not to throw them away. I expect they will be renamed and reused as new tests are written.